utils.cpp

```diff
@@ -337,34 +329,25 @@ HPHP::Object Utils::doExecuteQuery(const HPHP::String ns, mongoc_client_t *clien
 	}
 
 	/* Get query properties */
-	auto zquery = query->o_get(s_query, false, HPHP::s_MongoDriverQuery_className);
+	auto zfilter = query->o_get(s_filter, false, HPHP::s_MongoDriverQuery_className);
+	auto zopts   = query->o_get(s_opts, false, HPHP::s_MongoDriverQuery_className);
+	auto zrc     = query->o_get(s_readConcernLevel, false, HPHP::s_MongoDriverQuery_className);
 
-	if (zquery.getType() == HPHP::KindOfArray) {
-		const HPHP::Array& aquery = zquery.toArray();
+	HPHP::VariantToBsonConverter filter_converter(zfilter, HIPPO_BSON_NO_FLAGS);
+	bfilter = bson_new();
+	filter_converter.convert(bfilter);
+	
+	HPHP::VariantToBsonConverter opts_converter(zopts, HIPPO_BSON_NO_FLAGS);
+	bopts = bson_new();
+	opts_converter.convert(bopts);
 
-		skip = aquery[s_skip].toInt32();
-		limit = aquery[s_limit].toInt32();
-		batch_size = aquery[s_batchSize].toInt32();
-		flags = (mongoc_query_flags_t) aquery[s_flags].toInt32();
+	if (!zrc.isNull()) {
+		mongoc_read_concern_t *rc;
 
-		HPHP::VariantToBsonConverter converter(aquery[s_query], HIPPO_BSON_NO_FLAGS);
-		bson_query = bson_new();
-		converter.convert(bson_query);
-
-		if (aquery.exists(s_fields)) {
-			HPHP::VariantToBsonConverter converter(aquery[s_fields], HIPPO_BSON_NO_FLAGS);
-			bson_fields = bson_new();
-			converter.convert(bson_fields);
-		}
+		rc = mongoc_read_concern_new();
+		mongoc_read_concern_set_level(rc, zrc.toString().c_str());
 
-		if (aquery.exists(s_readConcern)) {
-			mongoc_read_concern_t *rc;
-
-			rc = mongoc_read_concern_new();
-			mongoc_read_concern_set_level(rc, aquery[s_readConcern].toString().c_str());
-
-			mongoc_client_set_read_concern(client, rc);
-		}
+		mongoc_client_set_read_concern(client, rc);
```

Comment:
I release the previous code was also setting the `mongoc_read_concern_t` on the `mongoc_client_t`, but shouldn't this be assigned to the `mongo_collection_t` object? As-is, I expect this might linger after the query is executed.

Apologies if I missed this in the original ReadConcern PR.